packages/cli/src/lib/bundler/config.ts

```diff
@@ -47,8 +47,12 @@ const BUILD_CACHE_ENV_VAR = 'BACKSTAGE_CLI_EXPERIMENTAL_BUILD_CACHE';
 
 export function resolveBaseUrl(config: Config): URL {
   const baseUrl = config.getOptionalString('app.baseUrl');
+
   try {
-    return new URL(baseUrl ?? '/', 'http://localhost:3000');
+    return new URL(
+      baseUrl ?? '/',
+      `http://localhost:${process.env.PORT ?? '3000'}`,
```

Comment:
I think it's best to limit `PORT` to only work when module federation is enabled, probably also only if we're in mode `'remote'`. The other option would be to fully support `PORT`, but afaik we'd need to rewrite the config that's being forwarded to the frontend and also make sure the backend is aware of the frontend port somehow, which is one of the reasons we don't support it right now.

Reply:
Done in 02434ddbabeb - probably passing the `moduleFederation` param in a couple unnecessary places, but I think it's probably best to avoid being too picky about when we pass it and don't, and strictly rely on whether we're in `'remote'` mode.